src/main/scala/io/citrine/lolo/trees/splits/Split.scala

```diff
@@ -41,7 +41,7 @@ class NoSplit extends Split {
     *
     * @return index of the input vector used by this split
     */
-  override def getIndex(): Int = -1
+  override def index: Int = -1
```

Comment:
Would it be more appropriate to have the index be an `Option[Int]` to account for the fact that a split may not have a valid index?

Probably a ton of work to update that now b/c of its widespread usages in the code, but seems like a more appropriate interface.

Reply:
Using -1 to mean "no split" isn't great. But I don't think an option would be better because then you allow a situation in which an actual split has no index, and `index` should _always_ be defined for split objects.

I think a better structure would be to have a `trait SplitResult` which is extended by `case class NoSplit extends SplitResult` and `trait Split` extends `SplitResult`. The trait `Split` would have an `index` and it would be extended by `RealSplit` and `CategoricalSplit`. So you could match on the `SplitResult` to establish whether the split was successful, and then you know that `index` is well-defined.

But that's a bit more than I want to take on right now.